linera-core/src/chain_worker/state/mod.rs

```diff
@@ -418,10 +418,22 @@ where
         let mut heights_by_recipient = BTreeMap::<_, BTreeMap<_, _>>::new();
         let mut targets = self.chain.outboxes.indices().await?;
         if let Some(tracked_chains) = self.tracked_chains.as_ref() {
+            let mut publishers = HashSet::new();
+            self.chain
+                .execution_state
+                .system
+                .subscriptions
+                .for_each_index(|subscription| {
+                    publishers.insert(subscription.chain_id);
+                    Ok(())
+                })
+                .await?;
```

Comment:
Not sure what's better, but this seems more declarative:
```suggestion
            let publishers = self
                .chain
                .execution_state
                .system
                .subscriptions
                .indices()
                .await?
                .iter()
                .map(|subscription| subscription.chain_id)
                .collect::<HashSet<_>>();
```